core/server/common/src/main/java/alluxio/master/journal/MetadataSyncMergeJournalContext.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.journal;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A {@link FileSystemMergeJournalContext} used in MetadataSync (InodeSyncStream).
+ *
+ * This journal context never enforces a synchronous journal flush but only appends the journals
+ * into the async journal writer. Metadata sync creates a bunch of inodes and with
+ * {@link alluxio.conf.PropertyKey#MASTER_FILE_SYSTEM_MERGE_INODE_JOURNALS} enabled,
+ * journals are committed as long as an inode path lock is released, which creates too many
+ * journal persistence/ raft RPCs and may impair the metadata sync performance.
+ *
+ * Hence, this class only flushes journals async on flush() and close() and the underlying
+ * journals in the async journal writer will be committed when the underlying JournalContext
+ * gets closed.
+ *
+ * Each metadata sync job should have its own
+ * {@link MetadataSyncMergeJournalContext} instance.
+ */
+@ThreadSafe
```

Comment:
I wonder if you need the synchronized notation on these methods? It is only calling the internal object, and the internal object is synchronized right?

Also there is one of these objects created for each thread right? So it is not called concurrently?

I guess its not a big deal either way, just trying to understand where the concurrency will happen.

Reply:
made it @NotThreadSafe